src/number/format-decimal.js

```diff
@@ -14,18 +14,24 @@ module.exports = function(value, localeData, options) {
 	// round to desired precision
 	value = Math.abs(Math.round(value * precisionScaling) / precisionScaling);
 
-	var integerValue = value | 0;
+	var integerValue = Math.trunc( value );
 
 	var ret = formatPositiveInteger(integerValue, localeData, options);
 
 	var hasDecimal = value !== integerValue;
 	var decimalStr = null;
 
 	if (hasDecimal) {
-		// get a string of 0.xxx
-		decimalStr = '' + (Math.round((value - integerValue) * precisionScaling) / precisionScaling);
-		// the first decimal place is index 2
-		decimalStr = decimalStr.slice(2);
+		var decimalValue = Math.round( (value - integerValue) * precisionScaling) / precisionScaling;
+		if (decimalValue.toExponential() === decimalValue.toString()) {
```

Comment:
We're `toString`ing multiple times now. Could this go back to `decimalStr = '' + (...)` and then `if (decimalStr === decimalValue.toExponential()) { formatExponentialDecimalString(decimalStr); }`